lib/nodes/opal_async.cpp

```diff
@@ -0,0 +1,416 @@
+/* Node type: OPAL-RT Asynchronous Process (libOpalAsyncApi)
+ *
+ * Author: Steffen Vogel <steffen.vogel@opal-rt.com>
+ * SPDX-FileCopyrightText: 2023-2025 OPAL-RT Germany GmbH
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <cstdlib>
+#include <thread>
+#include <vector>
+
+#include <unistd.h>
+
+#include <villas/exceptions.hpp>
+#include <villas/log.hpp>
+#include <villas/nodes/opal_async.hpp>
+#include <villas/utils.hpp>
+
+using namespace villas;
+using namespace villas::node;
+using namespace villas::node::opal;
+using namespace villas::utils;
+
+// This defines the maximum number of signals (doubles) that can be sent
+// or received by any individual OpAsyncSend/Recv blocks in the model. This
+// only applies to the "model <-> asynchronous process" communication.
+#define MAXSENDSIZE 64
+#define MAXRECVSIZE 64
+
+// Private static storage
+static std::string
+    shmemAsyncName; // Shared Memory identifiers and size, provided via argv or config.
+static std::string
+    shmemSystemCtrlName; // Shared Memory identifiers and size, provided via argv or config.
+static size_t
+    shmemAsyncSize; // Shared Memory identifiers and size, provided via argv or config.
```

Comment:
The line breaks are a bit weird. Consider moving the comment above the declarations.